### DISCUSSION TITLE
How to setup eslint with a custom-server?

### DISCUSSION THREAD

**Discussion by Floriferous** [vercel/next.js#68620] [Help]
### Summary

I'm using NextJS' custom server, starting from the example here: https://github.com/vercel/next.js/tree/canary/examples/custom-server

There are 2 tsconfig.json files in the same folder, a regular tsconfig, and tsconfig.server that extends the first one. 

I'm a bit puzzled how these multi-tsconfig folders work, and how well they play together. How do you configure eslint in this case? It needs a tsconfig project, but I can't give it 2 different files it seems, so either my server files are linted, or my client files. 

I will probably need 2 different eslint configs here, but I'm not sure how to do that. Does someone have a good pattern in mind for how to set this up?

### Example

https://github.com/vercel/next.js/tree/canary/examples/custom-server

**Comment by Juneezee** [comment]
You'll only need one ESLint configuration file. You use the `files` property to control which rules to apply for which files. For example

```js
// eslint.config.js
export default [
    {
        files: ["server.ts"],
        rules: {
            semi: "error",
            "prefer-const": "error"
        }
    },
    {
        files: ["**/*.ts"],
        ignores: ["server.ts"]
        rules: {
            ...
        }
    }
];
```

See more at https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file
